### ISSUE TITLE
Feature Request: get point of time in song when clicking on a note

### ISSUE THREAD

**Issue by josyjanga** [paulrosen/abcjs#270]
Hi Paul,
I couldn't find a nice way to accomplish the following:
when clicking on a beat, I need to set the point in time of the song to an external audio player. I couldn't find data for the progress in the click listener callback.

In Feature Request #202 you showed me how to set the progress with help of the animation api, but in this case, the I think the click callback needs to be extended to get the position in milliseconds or percentage of the beat I clicked on.

Maybe there is already a way to do this and I didn's see it. Thank you for your help in advance.
Keep up the great work :-)

**Comment by paulrosen** [comment]
When you say "clicking on a beat" are you clicking on a note in the visual display and handling the callback? Are you clicking on something in your own UI? Are you clicking on the playback control that I draw?

Are you using synth sound from abcjs? I'm guessing that you are using an external file for sound and using abcjs for showing the music and doing animation?

Did you upgrade to the latest abcjs? I have been improving this very area.

**Comment by josyjanga** [comment]
Yes, I mean clicking on a note in the visual display of abcjs:
<img width="344" alt="Bildschirmfoto 2019-11-08 um 18 32 36" src="https://user-images.githubusercontent.com/5549075/68498211-9779b400-0256-11ea-9381-7a66b7bbf0ef.png">

For example in your configuration tool: If you click on a note here https://configurator.abcjs.net/audio,  the audio player cursor should jump to that position you clicked. For this to work, I think the click listener callback would need to get extended.

I am using the sound generated with abjcs, but I am playing it with an external audio player (it's not possible to use the abcjs player for my usecase).

If a note is clicked, I would need information in milliseconds or percentage, at which position I am, so I can synchronize the external audio player. This should be a feature for when the song is not playing and not animating.
As fallback I could count the total beats in the song and check at which beat the clicked note is in for estimating the position.

I didn't upgrade to the latest abcjs yet, but I couldn't find anything in your documentation to get this to work.
Thanks.

**Comment by paulrosen** [comment]
Hi Elisabeth,

I finally got a minute to look at this. I recently made some improvements in this area and haven't had a chance to update the configurator yet. I have great plans for it!

With a recent version of abcjs, when you click a note you will get back an object that contains `currentTrackMilliseconds`. I hope you are able to upgrade abcjs. If not, let me know what problem you ran into when trying.

Note that if you click a note that has a repeat (i.e. `|:  :|`), it will give you the position of the first time through.

**Comment by josyjanga** [comment]
Hi Paul,

thank you for your help. I updated to the very latest version 5.10.2 and I already saw some nice new features. :-) 
But unfortunately I can't find the object which contains currentTrackMilliseconds.
I guess I'm kind of using the wrong listener.

I'm adding the clickListener like this:
`renderAbc('canvas', notation, {viewportHorizontal: true, add_classes: true, editable: false, staffwidth: width,  responsive: "resize", clickListener: clickedOnNote});`

But this click listener doesn't call anything in abcjs which updates `currentTrackMilliseconds`

The callback only gives me this data:
<img width="786" alt="Bildschirmfoto 2019-11-13 um 23 24 18" src="https://user-images.githubusercontent.com/5549075/68810492-9093dc80-066e-11ea-8b87-19e94bd6c730.png">

Is there another way to do this?
Thank you!

**Comment by paulrosen** [comment]
Sorry for the delay - I was at a conference.

Oh, yeah, that is only populated after you've used the new synth. You can see the example in the source code at `examples/full-synth.html`. If you immediately click on a note you get undefined. If you click the "play" button then it starts working.

Sorry about the misinformation.

I am planning a minor release this week. I will try to figure out something to do about this. I wonder if there is a way to support that without needing so support the creation of audio for people who don't need it.

**Comment by paulrosen** [comment]
I have a workaround for you. Right after your `renderAbc`, call this:
```
var midiBuffer = new ABCJS.synth.CreateSynth();
midiBuffer.init({ visualObj: visualObj });
```
Then your callback handler should work.

**Comment by josyjanga** [comment]
When I call currentTrackMilliseconds I get the sum of the played notes and not the time in milliseconds, for example I get 0.25, 0.75, 1.0, 1.5 etc. for currentTrackMilliseconds.

With your abc notation it just works fine, but with mine I don't get milliseconds.
You can test it yourself if you paste my notation to the abc variable in full-synth.html

`X:1\r\nC:Trad.\r\nL:1/4\r\nQ:1/4=180\r\nM:3/4\r\nK:C\r\n%%text Vorspiel\r\n[EG] |&quot;C&quot; [Ge]2 [Fd] |&quot;G7&quot; [Af]2 [FB] |\r\n&quot;C&quot; [Ec]3/2 [Ec]/ [Ec] | [Ec] z |!\r\n%%text Beginn des Tanzes\r\n[EG] |&quot;C&quot; [Ge]2 [Fd] |&quot;F&quot; [Ac]2 [FA] |\r\n&quot;C&quot; [EG]2 [CE] | [EG]2 [Ac] |\r\n&quot;G7&quot; [GB]2 [B,F] | [FG]2 [GB] |\r\n&quot;C&quot; [Ec]2 [CE] | [EG]2 [EG] |\r\n[Ge]2 [Fd] |&quot;F&quot; [Ac]2 [FA] |\r\n&quot;C&quot; [EG]2 [CE] | [EG]2 [Ac] |\r\n&quot;G7&quot; [GB]2 [FG] | [Af]2 [FB] |\r\n&quot;C&quot; [Ec]3/2 [Ec]/ [Ec] | [Ec] z [EG] |\r\n[Ge]2 [Fd] |&quot;F&quot; [Ac]2 [FA] |\r\n&quot;C&quot; [EG]2 [CE] | [EG]2 [Ac] |\r\n&quot;G7&quot; [GB]2 [B,F] | [FG]2 [GB] |\r\n&quot;C&quot; [Ec]2 [CE] | [EG]2 [EG] |\r\n[Ge]2 [Fd] |&quot;F&quot; [Ac]2 [FA] |\r\n&quot;C&quot; [EG]2 [CE] | [EG]2 [Ac] |\r\n&quot;G7&quot; [GB]2 [FG] | [Af]2 [FB] |\r\n&quot;C&quot; [Ec]3/2 [Ec]/ [Ec] | [Ec] z |]`

**Comment by paulrosen** [comment]
I duplicated this and see something interesting - I'll get a fix out soon.

**Comment by paulrosen** [comment]
This will be fixed in version 5.10.4, releaed soon.
